File: app/src/main/java/com/agrogames/islandsofwar/common/M.java
package com.agrogames.islandsofwar.common;

import com.agrogames.islandsofwar.engine.abs.common.Point;

public class M {
    public static int module(int a){
        return a >= 0 ? a : -a;
    }

    public static float module(float a){
        return a >= 0 ? a : -a;
    }

    public static double module(double a){
        return a >= 0 ? a : -a;
    }

    public static boolean nearlyEquals(float a, float b){
        return module(a - b) < 0.0001f;
    }
    public static float sqrt(float a){
        return (float) Math.sqrt(a);
    }

    public static float dist(Point a, Point b){
        float w = module(a.x - b.x);
        float h = module(a.y) - b.y;
        return sqrt(w * w + h * h);
    }

    public static Point middle(Point a, Point b){
        return new Point((a.x + b.x) / 2f, (a.y + b.y) / 2f);
    }
}

File: old/app/src/main/java/ru/agrogames/islands/common/M.kt
package ru.agrogames.islands.common

import ru.agrogames.islands.engine.abs.common.Point

object M {
    fun module(a: Int): Int {
        return if (a >= 0) a else -a
    }

    fun module(a: Float): Float {
        return if (a >= 0) a else -a
    }

    fun module(a: Double): Double {
        return if (a >= 0) a else -a
    }

    fun nearlyEquals(a: Float, b: Float): Boolean {
        return module(a - b) < 0.0001f
    }

    fun sqrt(a: Float): Float {
        return Math.sqrt(a.toDouble()).toFloat()
    }

    fun dist(a: Point, b: Point): Float {
        val w = module(a.x - b.x)
        val h = module(a.y) - b.y
        return sqrt(w * w + h * h)
    }

    fun middle(a: Point, b: Point): Point {
        return Point((a.x + b.x) / 2f, (a.y + b.y) / 2f)
    }
}